corehq/apps/locations/dbaccessors.py

```diff
@@ -92,3 +92,23 @@ def user_ids_at_accessible_locations(domain_name, user):
     from corehq.apps.locations.models import SQLLocation
     accessible_location_ids = SQLLocation.active_objects.accessible_location_ids(domain_name, user)
     return user_ids_at_locations(accessible_location_ids)
+
+
+def assigned_location_by_user_id(domain, location_ids):
```

Comment:
Ah. I see how this was confusing before. The User ES mock was incorrect. I fixed that https://github.com/dimagi/commcare-hq/pull/15018/commits/7ae622b4e116868cd71c74f32a047f4ccdc4d3e3

and changed this to use primary location https://github.com/dimagi/commcare-hq/pull/15018/commits/d85e72200d3fbba333412c14ef23f0d693339a55